### ISSUE TITLE
Feature request for "Reporter Role"

### ISSUE THREAD

**Issue by derkan** [gogits/gogs#5082]
- Gogs version (or commit ref): 0.11.34.1122
- Git version: 2.10.2
- Operating system: CentOS7 
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist (usually found in `log/gogs.log`):

## Description

This a **feature request** for defining a new role named as `Reporter`. This role owner will be able to add/view **İssues**, view only **Wiki**,  but will not be able to see or update private repository code and pull requests. This role is important to make internal or external users to add/follow issues to private repositories themselves.

**Comment by herol64** [comment]
We need this feature too!

**Comment by ekkodur** [comment]
This will be nice feature for Gogs.

**Comment by derkan** [comment]
Hi @Unknwon ,
Any plans to implement this feature?

**Comment by Unknwon** [comment]
@derkan this is not a standalone feature, Gogs should first support role-based model like GitLab to be able to add builtin and custom roles like this one.

**Comment by derkan** [comment]
Thanks for response. Actuall a RBAC like role is not needed. Currently `Collaboration` settings for a repo gives `Admin`, `Read`, `Write` options to be defined to a user. A `Reporter` option is needed here to hide source code but enable end users to submit issues, download releases and read wiki as manual. That will be much like `Read` role without source code access.

**Comment by Unknwon** [comment]
@derkan you can already have private repository with public issue/wiki in Gogs.

**Comment by derkan** [comment]
Thats right, but I want to have `Reporter` role for private repositories to make it limited accessible for end users.

**Comment by derkan** [comment]
I've checked Gitlab. And released that what I want is `Guest` role. Please check [here](https://docs.gitlab.com/ee/user/permissions.html). I think Gitlab is also using static roles(no way to define new roles).  As far as I see, Gogs needs `Guest` role.(not reporter, I'll rename title).
Regards

**Comment by Unknwon** [comment]
GitLab definitely has complete permission modal other than just simple **admin, read and write**. https://docs.gitlab.com/ee/user/permissions.html

But I can priority issue see if I can come up a basic foundation of RBAC.
